benchmarks/_benchmark.py

```diff
@@ -47,9 +48,60 @@ def arrow_info() -> Dict[str, Any]:
     }
 
 
+@dataclass
+class BenchmarkResult:
+    """
+    A dataclass for containing results from running a benchmark. Can be serialized to a
+    dict (and then on to JSON) with `dataclasses.asdict()`.
+
+    Attributes
+    ----------
+    name : str
+        benchmark name
+    run_id : str
+        ID string for the run, typically generated by conbench
+    batch_id : str
+        ID string for the batch, generated by conbench unless manually specified
+    result : Dict[str, Any]
+        Actual measurement data. Currently of the form ``{"data": <<data>>, "unit": "s"}``
+    params : Dict[str, Any]
+        Parameter names and arguments (i.e. the case) supplied for the run. Currently a
+        subset of ``tags``, but often needed separately.
+    tags : Dict[str, Any]
+        Many things. Determines history runs
+    info : Dict[str, Any]
+        Things like ``arrow_version``, ``arrow_compiler_id``, ``arrow_compiler_version``,
+        ``benchmark_language_version, ``arrow_version_r``
+    context : Dict[str, Any]
+        ``arrow_compiler_flags``, ``benchmark_language``
+    github : Dict[str, Any]
+        ``repository``, ``commit``
+    options : Dict[str, Any]
+        For R benchmarks, ``cpu_count``, ``iterations``, ``drop_caches``
+    output : str
+        stdout from process running the benchmark
+    error : str
+        stderr from process running the benchmark
+    """
```

Comment:
In addition to these fields, Benchmark result JSON sent to conbench has these fields:
- run_name
- run_reason
- machine_info or cluster_info
- timestamp

Should BenchmarkResult class also have these fields?